### ISSUE TITLE
cf_service_exists does not look for services in the current space only

### ISSUE THREAD

**Issue by FlorianSW** [patrickcrocker/cf-cli-resource#18]
The `cf_service_exists`, which is used by `cf_wait_for_delete_service_instance` to check, if the service is already deleted, does not limit the api request to one specific space, but instead it uses the "global" /v2/service_instances api endpoint (which lists all available service instances for the currently logged in user). That results in the wait_for_service_deleting property to wait for all services with the same name (even in other spaces) to be deleted, too, however, they're not deleted, so the wait results in a timeout.

Instead, it should use the specific URL for getting the service instances for the current space only. This URL can be retrieved with the /v2/spaces api endpoint, e.g. with this code:

```
local serviceInstancesUrl=`cf curl /v2/spaces?q=name:$currentSpace | jq -er '.resources[0].entity.service_instances_url'`
CF_TRACE=false cf curl $serviceInstancesUrl | jq -e --arg name "$service_instance" '.resources[] | select(.entity.name == $name) | true' >/dev/null
```

Where `$currentSpace` is the currently selected space.

**Comment by FlorianSW** [comment]
*yay* Cool, thanks for fixing! :)

**Comment by patrickcrocker** [comment]
np! this fix is now available in version 2.4.2/latest
